{
    "version": "https://jsonfeed.org/version/1",
    "title": "Security Advisory for Python packages hosted at PyPI.org",
    "home_page_url": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip",
    "feed_url": "https://azu.github.io/github-advisory-database-rss/pip.json",
    "description": "Security Advisory for Python packages hosted at PyPI.org on GitHub",
    "items": [
        {
            "content_html": "<p>sqlite-vec v0.1.1 was discovered to contain a heap buffer overflow via the npy_token_next function. This vulnerability allows attackers to cause a Denial of Service (DoS) via a crafted file.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-46488\">https://nvd.nist.gov/vuln/detail/CVE-2024-46488</a></li>\n<li><a href=\"https://github.com/VulnSphere/LLMVulnSphere/blob/main/VectorDB/sqlite-vec/OOBR_2.md\">https://github.com/VulnSphere/LLMVulnSphere/blob/main/VectorDB/sqlite-vec/OOBR_2.md</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-vrcx-gx3g-j3h8\">https://github.com/advisories/GHSA-vrcx-gx3g-j3h8</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-vrcx-gx3g-j3h8",
            "title": "[sqlite-vec] Heap-based Buffer Overflow in sqlite-vec",
            "date_modified": "2024-09-25T22:13:22.000Z",
            "date_published": "2024-09-25T18:31:21.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-vrcx-gx3g-j3h8"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>This vulnerability allows users of Gradio applications that have a public link (such as on Hugging Face Spaces) to access files on the machine hosting the Gradio application. This involves intercepting and modifying the network requests made by the Gradio app to the server. </p>\n<h3 id=\"patches\">Patches</h3>\n<p>Yes, the problem has been patched in Gradio version 4.19.2 or higher. We have no knowledge of this exploit being used against users of Gradio applications, but we encourage all users to upgrade to Gradio 4.19.2 or higher.</p>\n<p>Fixed in: <a href=\"https://github.com/gradio-app/gradio/commit/16fbe9cd0cffa9f2a824a0165beb43446114eec7\">https://github.com/gradio-app/gradio/commit/16fbe9cd0cffa9f2a824a0165beb43446114eec7</a>\nCVE: <a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-1728\">https://nvd.nist.gov/vuln/detail/CVE-2024-1728</a></p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/gradio-app/gradio/security/advisories/GHSA-m842-4qm8-7gpq\">https://github.com/gradio-app/gradio/security/advisories/GHSA-m842-4qm8-7gpq</a></li>\n<li><a href=\"https://github.com/gradio-app/gradio/commit/16fbe9cd0cffa9f2a824a0165beb43446114eec7\">https://github.com/gradio-app/gradio/commit/16fbe9cd0cffa9f2a824a0165beb43446114eec7</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-m842-4qm8-7gpq\">https://github.com/advisories/GHSA-m842-4qm8-7gpq</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-m842-4qm8-7gpq",
            "title": "[gradio] Gradio allows users to access arbitrary files",
            "date_modified": "2024-09-25T21:48:26.000Z",
            "date_published": "2024-09-25T21:48:24.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-m842-4qm8-7gpq"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p><em>What kind of vulnerability is it? Who is impacted?</em></p>\n<p>All users of Nautobot versions earlier than 1.5.7 are impacted.</p>\n<p>In Nautobot 1.5.7 we have enabled sandboxed environments for the Jinja2 template engine used internally for template rendering for the following objects:</p>\n<ul>\n<li><code>extras.ComputedField</code></li>\n<li><code>extras.CustomLink</code></li>\n<li><code>extras.ExportTemplate</code> </li>\n<li><code>extras.Secret</code></li>\n<li><code>extras.Webhook</code></li>\n</ul>\n<p>While we are not aware of any active exploits, we have made this change as a preventative measure to protect against any potential remote code execution attacks utilizing maliciously crafted template code.</p>\n<p>This change forces the Jinja2 template engine to use a <a href=\"https://jinja.palletsprojects.com/en/3.0.x/sandbox/#sandbox\"><code>SandboxedEnvironment</code></a> on all new installations of Nautobot.</p>\n<p>This addresses any potential unsafe code execution everywhere the helper function <code>nautobot.utilities.utils.render_jinja2</code> is called. Additionally, our documentation that was previously suggesting the direct use of <code>jinja2.Template</code> has been revised to utilize <code>render_jinja2</code>.</p>\n<h3 id=\"patches\">Patches</h3>\n<p><em>Has the problem been patched? What versions should users upgrade to?</em></p>\n<p>Yes. Users should upgrade to Nautobot 1.5.7 or newer.</p>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p><em>Is there a way for users to fix or remediate the vulnerability without upgrading?</em></p>\n<h5 id=\"enabling-sandboxed-environments\">Enabling Sandboxed Environments</h5>\n<p>For users that are unable to upgrade to the latest release of Nautobot, you may add the following setting to your <code>nautobot_config.py</code> to apply the sandbox environment enforcement:</p>\n<pre><code class=\"language-python\">TEMPLATES[1][\"OPTIONS\"][\"environment\"] = \"jinja2.sandbox.SandboxedEnvironment\"\n</code></pre>\n<p>After applying this change, you must restart all Nautobot services, including any Celery worker processes.</p>\n<p><strong>Note:</strong> <em>Nautobot specifies two template engines by default, the first being “django” for the Django built-in template engine, and the second being “jinja” for the Jinja2 template engine. This recommended setting will update the second item in the list of template engines, which is the Jinja2 engine.</em></p>\n<h5 id=\"restricting-jinja2-using-access-controls\">Restricting Jinja2 using Access Controls</h5>\n<p>For users that are unable to immediately update their configuration such as if a Nautobot service restart is too disruptive to operations, access to provide custom Jinja2 template values may be mitigated using permissions to restrict “change” (write) actions to the affected object types listed in the first section.</p>\n<p><strong>Note:</strong> <em>This solution is intended to be stopgap until you can successfully update your <code>nautobot_config.py</code> or upgrade your Nautobot instance to apply the sandboxed environment enforcement.</em></p>\n<h4 id=\"updating-existing-app-or-job-code\">Updating Existing App or Job Code</h4>\n<p>For Nautobot App (formerly plugin) authors or Job authors, additionally we recommend that if you have any custom code that may for example be using <code>jinaj2.Template</code> that you no longer use that. Instead, please always use our <code>nautobot.utilities.utils.render_jinja2</code> function which will make sure that the centrally-provided Jinja2 template engine with sandboxing enforced is being utilized.</p>\n<p>Anywhere you’ve been using this pattern:</p>\n<pre><code class=\"language-python\">from jinja2 import Template\n\nmy_template = Template(template_code)\nconfig = my_template.render(context)\n</code></pre>\n<p>We recommend that you replace it with this pattern:</p>\n<pre><code class=\"language-python\">from nautobot.utilities.utils import render_jinja2\n    \nconfig = render_jinja2(template_code, context)\n</code></pre>\n<h3 id=\"references\">References</h3>\n<p><em>Are there any links users can visit to find out more?</em></p>\n<p>Please see the Nautobot 1.5.7 release notes. </p>\n<p><a href=\"https://docs.nautobot.com/projects/core/en/stable/release-notes/version-1.5/#v157-2023-01-04\">https://docs.nautobot.com/projects/core/en/stable/release-notes/version-1.5/#v157-2023-01-04</a></p>\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/nautobot/nautobot/security/advisories/GHSA-8mfq-f5wj-vw5m\">https://github.com/nautobot/nautobot/security/advisories/GHSA-8mfq-f5wj-vw5m</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-25657\">https://nvd.nist.gov/vuln/detail/CVE-2023-25657</a></li>\n<li><a href=\"https://github.com/nautobot/nautobot/commit/d47f157e83b0c353bb2b697f911882c71cf90ca0\">https://github.com/nautobot/nautobot/commit/d47f157e83b0c353bb2b697f911882c71cf90ca0</a></li>\n<li><a href=\"https://docs.nautobot.com/projects/core/en/stable/release-notes/version-1.5/#v157-2023-01-04\">https://docs.nautobot.com/projects/core/en/stable/release-notes/version-1.5/#v157-2023-01-04</a></li>\n<li><a href=\"https://jinja.palletsprojects.com/en/3.0.x/sandbox/#sandbox\">https://jinja.palletsprojects.com/en/3.0.x/sandbox/#sandbox</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/nautobot/PYSEC-2023-37.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/nautobot/PYSEC-2023-37.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-8mfq-f5wj-vw5m\">https://github.com/advisories/GHSA-8mfq-f5wj-vw5m</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-8mfq-f5wj-vw5m",
            "title": "[nautobot] Nautobot vulnerable to remote code execution via Jinja2 template rendering",
            "date_modified": "2024-09-25T20:47:41.000Z",
            "date_published": "2023-02-22T00:04:02.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-8mfq-f5wj-vw5m"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>When using the Linux bridge ml2 driver, non-privileged tenants are able to create and attach ports without specifying an IP address, bypassing IP address validation. A potential denial of service could occur if an IP address, conflicting with existing guests or routers, is then assigned from outside of the allowed allocation pool. Versions of openstack-neutron before 13.0.0.0b2, 12.0.3 and 11.0.5 are vulnerable.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2018-14635\">https://nvd.nist.gov/vuln/detail/CVE-2018-14635</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2710\">https://access.redhat.com/errata/RHSA-2018:2710</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2715\">https://access.redhat.com/errata/RHSA-2018:2715</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2721\">https://access.redhat.com/errata/RHSA-2018:2721</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:3792\">https://access.redhat.com/errata/RHSA-2018:3792</a></li>\n<li><a href=\"https://bugs.launchpad.net/neutron/+bug/1757482\">https://bugs.launchpad.net/neutron/+bug/1757482</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-14635\">https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-14635</a></li>\n<li><a href=\"https://git.openstack.org/cgit/openstack/neutron/commit/?id=54aa6e81cb17b33ce4d5d469cc11dec2869c762d\">https://git.openstack.org/cgit/openstack/neutron/commit/?id=54aa6e81cb17b33ce4d5d469cc11dec2869c762d</a></li>\n<li><a href=\"https://github.com/openstack/neutron/commit/54aa6e81cb17b33ce4d5d469cc11dec2869c762d\">https://github.com/openstack/neutron/commit/54aa6e81cb17b33ce4d5d469cc11dec2869c762d</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/neutron/PYSEC-2018-93.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/neutron/PYSEC-2018-93.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-x634-34m9-96mp\">https://github.com/advisories/GHSA-x634-34m9-96mp</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-x634-34m9-96mp",
            "title": "[neutron] OpensStack Neutron Denial of Service Vulnerability",
            "date_modified": "2024-09-25T20:45:22.000Z",
            "date_published": "2022-05-13T01:07:33.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-x634-34m9-96mp"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>When using the Linux bridge ml2 driver, non-privileged tenants are able to create and attach ports without specifying an IP address, bypassing IP address validation. A potential denial of service could occur if an IP address, conflicting with existing guests or routers, is then assigned from outside of the allowed allocation pool. Versions of openstack-neutron before 13.0.0.0b2, 12.0.3 and 11.0.5 are vulnerable.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2018-14635\">https://nvd.nist.gov/vuln/detail/CVE-2018-14635</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2710\">https://access.redhat.com/errata/RHSA-2018:2710</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2715\">https://access.redhat.com/errata/RHSA-2018:2715</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2721\">https://access.redhat.com/errata/RHSA-2018:2721</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:3792\">https://access.redhat.com/errata/RHSA-2018:3792</a></li>\n<li><a href=\"https://bugs.launchpad.net/neutron/+bug/1757482\">https://bugs.launchpad.net/neutron/+bug/1757482</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-14635\">https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-14635</a></li>\n<li><a href=\"https://git.openstack.org/cgit/openstack/neutron/commit/?id=54aa6e81cb17b33ce4d5d469cc11dec2869c762d\">https://git.openstack.org/cgit/openstack/neutron/commit/?id=54aa6e81cb17b33ce4d5d469cc11dec2869c762d</a></li>\n<li><a href=\"https://github.com/openstack/neutron/commit/54aa6e81cb17b33ce4d5d469cc11dec2869c762d\">https://github.com/openstack/neutron/commit/54aa6e81cb17b33ce4d5d469cc11dec2869c762d</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/neutron/PYSEC-2018-93.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/neutron/PYSEC-2018-93.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-x634-34m9-96mp\">https://github.com/advisories/GHSA-x634-34m9-96mp</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-x634-34m9-96mp",
            "title": "[neutron] OpensStack Neutron Denial of Service Vulnerability",
            "date_modified": "2024-09-25T20:45:22.000Z",
            "date_published": "2022-05-13T01:07:33.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-x634-34m9-96mp"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A Denial of Service (DoS) condition is possible in OpenStack Mistral in versions up to and including 7.0.3. Submitting a specially crafted workflow definition YAML file containing nested anchors can lead to resource exhaustion culminating in a denial of service.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2018-16848\">https://nvd.nist.gov/vuln/detail/CVE-2018-16848</a></li>\n<li><a href=\"https://bugs.launchpad.net/mistral/+bug/1785657\">https://bugs.launchpad.net/mistral/+bug/1785657</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=1645332\">https://bugzilla.redhat.com/show_bug.cgi?id=1645332</a></li>\n<li><a href=\"https://github.com/openstack/mistral/commit/eac23d9e774f658f9d4743c99aa2743eb104c3f9\">https://github.com/openstack/mistral/commit/eac23d9e774f658f9d4743c99aa2743eb104c3f9</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/mistral/PYSEC-2020-240.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/mistral/PYSEC-2020-240.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-443j-6p7g-6v4w\">https://github.com/advisories/GHSA-443j-6p7g-6v4w</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-443j-6p7g-6v4w",
            "title": "[mistral] OpenStack Mistral DoS",
            "date_modified": "2024-09-25T20:28:05.000Z",
            "date_published": "2022-05-24T17:20:25.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-443j-6p7g-6v4w"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>In Mistune through 2.0.2, support of inline markup is implemented by using regular expressions that can involve a high amount of backtracking on certain edge cases. This behavior is commonly named catastrophic backtracking.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-34749\">https://nvd.nist.gov/vuln/detail/CVE-2022-34749</a></li>\n<li><a href=\"https://github.com/lepture/mistune/commit/a6d43215132fe4f3d93f8d7e90ba83b16a0838b2\">https://github.com/lepture/mistune/commit/a6d43215132fe4f3d93f8d7e90ba83b16a0838b2</a></li>\n<li><a href=\"https://github.com/lepture/mistune/releases\">https://github.com/lepture/mistune/releases</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/mistune/PYSEC-2022-237.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/mistune/PYSEC-2022-237.yaml</a></li>\n<li><a href=\"https://github.com/lepture/mistune/issues/314#issuecomment-1223972386\">https://github.com/lepture/mistune/issues/314#issuecomment-1223972386</a></li>\n<li><a href=\"https://github.com/lepture/mistune/commit/ca1e7b506850f4e488823fc7338b49a8f9852718\">https://github.com/lepture/mistune/commit/ca1e7b506850f4e488823fc7338b49a8f9852718</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/TQHXITQ2DSBYOILKHXBSBB7PFBPZHF63\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/TQHXITQ2DSBYOILKHXBSBB7PFBPZHF63</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-fw3v-x4f2-v673\">https://github.com/advisories/GHSA-fw3v-x4f2-v673</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-fw3v-x4f2-v673",
            "title": "[mistune] Mistune vulnerable to catastrophic backtracking",
            "date_modified": "2024-09-25T20:07:28.000Z",
            "date_published": "2022-07-26T00:00:27.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-fw3v-x4f2-v673"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A vulnerability was found in mod_wsgi. The X-Client-IP header is not removed from a request from an untrusted proxy, allowing an attacker to pass the X-Client-IP header to the target WSGI application because the condition to remove it is missing.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-2255\">https://nvd.nist.gov/vuln/detail/CVE-2022-2255</a></li>\n<li><a href=\"https://github.com/GrahamDumpleton/mod_wsgi/blob/4.9.2/src/server/mod_wsgi.c#L13940-L13941\">https://github.com/GrahamDumpleton/mod_wsgi/blob/4.9.2/src/server/mod_wsgi.c#L13940-L13941</a></li>\n<li><a href=\"https://github.com/GrahamDumpleton/mod_wsgi/blob/4.9.2/src/server/mod_wsgi.c#L14046-L14082\">https://github.com/GrahamDumpleton/mod_wsgi/blob/4.9.2/src/server/mod_wsgi.c#L14046-L14082</a></li>\n<li><a href=\"https://modwsgi.readthedocs.io/en/latest/release-notes/version-4.9.3.html\">https://modwsgi.readthedocs.io/en/latest/release-notes/version-4.9.3.html</a></li>\n<li><a href=\"https://github.com/GrahamDumpleton/mod_wsgi/commit/af3c0c2736bc0b0b01fa0f0aad3c904b7fa9c751\">https://github.com/GrahamDumpleton/mod_wsgi/commit/af3c0c2736bc0b0b01fa0f0aad3c904b7fa9c751</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/mod-wsgi/PYSEC-2022-254.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/mod-wsgi/PYSEC-2022-254.yaml</a></li>\n<li><a href=\"https://lists.debian.org/debian-lts-announce/2022/09/msg00021.html\">https://lists.debian.org/debian-lts-announce/2022/09/msg00021.html</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-7527-8855-9cf8\">https://github.com/advisories/GHSA-7527-8855-9cf8</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-7527-8855-9cf8",
            "title": "[mod-wsgi] Incorrect header handling in mod-wsgi",
            "date_modified": "2024-09-25T20:02:58.000Z",
            "date_published": "2022-08-26T00:03:30.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-7527-8855-9cf8"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Cross-site Scripting (XSS) - Reflected in GitHub repository modoboa/modoboa prior to 2.0.45.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-0949\">https://nvd.nist.gov/vuln/detail/CVE-2023-0949</a></li>\n<li><a href=\"https://github.com/modoboa/modoboa/commit/aa74e9a4a870162eea169e0a6a2eab841f8811b7\">https://github.com/modoboa/modoboa/commit/aa74e9a4a870162eea169e0a6a2eab841f8811b7</a></li>\n<li><a href=\"https://huntr.dev/bounties/ef87be4e-493b-4ee9-9738-44c55b8acc19\">https://huntr.dev/bounties/ef87be4e-493b-4ee9-9738-44c55b8acc19</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/modoboa/PYSEC-2023-33.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/modoboa/PYSEC-2023-33.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-mgmm-cmhj-2h5f\">https://github.com/advisories/GHSA-mgmm-cmhj-2h5f</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-mgmm-cmhj-2h5f",
            "title": "[modoboa] modoboa Cross-site Scripting vulnerability",
            "date_modified": "2024-09-25T20:00:48.000Z",
            "date_published": "2023-02-22T09:30:18.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-mgmm-cmhj-2h5f"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Git before 1.8.5.6, 1.9.x before 1.9.5, 2.0.x before 2.0.5, 2.1.x before 2.1.4, and 2.2.x before 2.2.1 on Windows and OS X; Mercurial before 3.2.3 on Windows and OS X; Apple Xcode before 6.2 beta 3; mine; libgit2; Egit; and JGit allow remote Git servers to execute arbitrary commands via a tree containing a crafted .git/config file with (1) an ignorable Unicode codepoint, (2) a git~1/config representation, or (3) mixed case that is improperly handled on a case-insensitive filesystem.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2014-9390\">https://nvd.nist.gov/vuln/detail/CVE-2014-9390</a></li>\n<li><a href=\"https://github.com/libgit2/libgit2/commit/928429c5c96a701bcbcafacb2421a82602b36915\">https://github.com/libgit2/libgit2/commit/928429c5c96a701bcbcafacb2421a82602b36915</a></li>\n<li><a href=\"https://github.com/blog/1938-git-client-vulnerability-announced\">https://github.com/blog/1938-git-client-vulnerability-announced</a></li>\n<li><a href=\"https://news.ycombinator.com/item?id=8769667\">https://news.ycombinator.com/item?id=8769667</a></li>\n<li><a href=\"http://article.gmane.org/gmane.linux.kernel/1853266\">http://article.gmane.org/gmane.linux.kernel/1853266</a></li>\n<li><a href=\"http://git-blame.blogspot.com/2014/12/git-1856-195-205-214-and-221-and.html\">http://git-blame.blogspot.com/2014/12/git-1856-195-205-214-and-221-and.html</a></li>\n<li><a href=\"http://mercurial.selenic.com/wiki/WhatsNew\">http://mercurial.selenic.com/wiki/WhatsNew</a></li>\n<li><a href=\"http://support.apple.com/kb/HT204147\">http://support.apple.com/kb/HT204147</a></li>\n<li><a href=\"https://projects.eclipse.org/projects/technology.jgit/releases/3.5.3\">https://projects.eclipse.org/projects/technology.jgit/releases/3.5.3</a></li>\n<li><a href=\"https://web.archive.org/web/20211204220400/https://securitytracker.com/id?1031404\">https://web.archive.org/web/20211204220400/https://securitytracker.com/id?1031404</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/mercurial/PYSEC-2020-217.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/mercurial/PYSEC-2020-217.yaml</a></li>\n<li><a href=\"https://libgit2.org/security\">https://libgit2.org/security</a></li>\n<li><a href=\"http://securitytracker.com/id?1031404\">http://securitytracker.com/id?1031404</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-6vvc-c2m3-cjf3\">https://github.com/advisories/GHSA-6vvc-c2m3-cjf3</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-6vvc-c2m3-cjf3",
            "title": "[mercurial] JGit Improper Input Validation vulnerability",
            "date_modified": "2024-09-25T19:55:48.000Z",
            "date_published": "2022-05-17T19:57:29.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-6vvc-c2m3-cjf3"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"summary\">Summary</h3>\n<p>An unsafe extraction is being performed using <code>tarfile.extractall()</code> from a remotely retrieved tarball. Which may lead to the writing of the extracted files to an unintended location. Sometimes, the vulnerability is called a TarSlip or a ZipSlip variant.</p>\n<h3 id=\"details\">Details</h3>\n<p>I commented the following snippet of code as a vulnerability details. The code is from <a href=\"https://github.com/mindsdb/mindsdb/blob/afedd37c16e579b6dc075b0814e42d0505ccdc07/mindsdb/api/http/namespaces/file.py#L26..L134\">file.py#L26..L134</a></p>\n<pre><code class=\"language-python\">@ns_conf.route('/&lt;name&gt;')\n@ns_conf.param('name', \"MindsDB's name for file\")\nclass File(Resource):\n    @ns_conf.doc('put_file')\n    def put(self, name: str):\n        ''' add new file\n            params in FormData:\n                - file\n                - original_file_name [optional]\n        '''\n\n        data = {}\n\n        ... omitted for brevity\n\n            url = data['source']\n            data['file'] = data['name']\n\n            ... omitted for brevity \n\n            with requests.get(url, stream=True) as r:                   # Source: retrieve the URL which point to a remotely located tarball \n                if r.status_code != 200:\n                    return http_error(\n                        400,\n                        \"Error getting file\",\n                        f\"Got status code: {r.status_code}\"\n                    )\n                file_path = os.path.join(temp_dir_path, data['file'])\n                with open(file_path, 'wb') as f:\n                    for chunk in r.iter_content(chunk_size=8192):   # write with chunks the remote retrieved file into file_path location \n                        f.write(chunk)\n\n        original_file_name = data.get('original_file_name')\n\n        file_path = os.path.join(temp_dir_path, data['file'])      \n        lp = file_path.lower()\n        if lp.endswith(('.zip', '.tar.gz')):\n            if lp.endswith('.zip'):\n                with zipfile.ZipFile(file_path) as f:\n                    f.extractall(temp_dir_path)\n            elif lp.endswith('.tar.gz'):\n                with tarfile.open(file_path) as f:  # Just after \n                    f.extractall(temp_dir_path)  # Sink: the tarball located by file_path is supposed to be extracted to temp_dir_path. \n</code></pre>\n<p>So, a remotely available tarball is being retrieved and written to the server filesystem in chunks, and then, if the extension ends with <code>.tar.gz</code> of a compressed tarball, the mindsdb app applies <code>tarfile.extractall()</code> directly with no checks for the destination. </p>\n<p>However, according to the following <a href=\"https://docs.python.org/3/library/tarfile.html#tarfile.TarFile.extractall\">warning</a> from the official documentation;</p>\n<blockquote>\n<p>Warning: Never extract archives from untrusted sources without prior inspection. It is possible that files are created outside of path, e.g. members that have absolute filenames starting with \"/\" or filenames with two dots \"..\". </p>\n</blockquote>\n<h3 id=\"poc\">PoC</h3>\n<p>The following PoC is provided for illustration purposes only. It showcases the risk of extracting a non-harmless text file <code>sim4n6.txt</code> to one of the parent locations rather than the intended current folder.</p>\n<pre><code class=\"language-bash\">&gt; tar --list -v -f archive.tar.gz\ntar: Removing leading \"../../../\" from member names\n../../../sim4n6.txt\n\n&gt; python3 \nPython 3.10.6 (main, Nov  2 2022, 18:53:38) [GCC 11.3.0] on linux\nType \"help\", \"copyright\", \"credits\" or \"license\" for more information.\n&gt;&gt;&gt; import tarfile\n&gt;&gt;&gt; with tarfile.open(\"archive.tar.gz\") as tf:\n&gt;&gt;&gt;         tf.extractall()\n&gt;&gt;&gt; exit()\n\n&gt; test -f ../../../sim4n6.txt &amp;&amp; echo \"sim4n6.txt exists\"\nsim4n6.txt exists\n</code></pre>\n<h3 id=\"attack-scenario\">Attack Scenario</h3>\n<p>An attacker could craft a malicious tarball with a filename path, such as ../../../../../../../../etc/passwd, and then serve the archive remotely, proceed to the PUT request of the tarball through mindsdb and overwrite the system files of the hosting server for instance.</p>\n<h3 id=\"mitigation\">Mitigation</h3>\n<p>Potential mitigation could be to:</p>\n<ul>\n<li>Use a safer module, like <code>zipfile</code>.</li>\n<li>Use an alternative of <code>tarfile</code>, such as <code>tarsafe</code>. </li>\n<li>Validate the location or the absolute path of the extracted files and discard those with malicious paths such as relative path <code>../../..</code> or absolute path such as <code>/etc/password</code>. A simple wrapper could be written to raise an exception when a path traversal may be identified.</li>\n</ul>\n<p>This is similar to the other report <a href=\"https://github.com/mindsdb/mindsdb/security/advisories/GHSA-7x45-phmr-9wqp\">GHSA-7x45-phmr-9wqp</a>.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/mindsdb/mindsdb/security/advisories/GHSA-2g5w-29q9-w6hx\">https://github.com/mindsdb/mindsdb/security/advisories/GHSA-2g5w-29q9-w6hx</a></li>\n<li><a href=\"https://github.com/mindsdb/mindsdb/blob/afedd37c16e579b6dc075b0814e42d0505ccdc07/mindsdb/api/http/namespaces/file.py#L26..L134\">https://github.com/mindsdb/mindsdb/blob/afedd37c16e579b6dc075b0814e42d0505ccdc07/mindsdb/api/http/namespaces/file.py#L26..L134</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-30620\">https://nvd.nist.gov/vuln/detail/CVE-2023-30620</a></li>\n<li><a href=\"https://github.com/mindsdb/mindsdb/commit/4419b0f0019c000db390b54d8b9d06e1d3670039\">https://github.com/mindsdb/mindsdb/commit/4419b0f0019c000db390b54d8b9d06e1d3670039</a></li>\n<li><a href=\"https://github.com/mindsdb/mindsdb/releases/tag/v23.2.1.0\">https://github.com/mindsdb/mindsdb/releases/tag/v23.2.1.0</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/mindsdb/PYSEC-2023-27.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/mindsdb/PYSEC-2023-27.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-2g5w-29q9-w6hx\">https://github.com/advisories/GHSA-2g5w-29q9-w6hx</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-2g5w-29q9-w6hx",
            "title": "[mindsdb] mindsdb arbitrary file write when extracting a remotely retrieved Tarball",
            "date_modified": "2024-09-25T19:46:33.000Z",
            "date_published": "2023-03-30T21:42:48.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-2g5w-29q9-w6hx"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A vulnerability was found in MicroPython 1.23.0. It has been classified as critical. Affected is the function mp_vfs_umount of the file extmod/vfs.c of the component VFS Unmount Handler. The manipulation leads to heap-based buffer overflow. It is possible to launch the attack remotely. The exploit has been disclosed to the public and may be used. The name of the patch is 29943546343c92334e8518695a11fc0e2ceea68b. It is recommended to apply a patch to fix this issue. In the VFS unmount process, the comparison between the mounted path string and the unmount requested string is based solely on the length of the unmount string, which can lead to a heap buffer overflow read.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-8946\">https://nvd.nist.gov/vuln/detail/CVE-2024-8946</a></li>\n<li><a href=\"https://github.com/micropython/micropython/issues/13006\">https://github.com/micropython/micropython/issues/13006</a></li>\n<li><a href=\"https://github.com/micropython/micropython/issues/13006#issuecomment-1820309455\">https://github.com/micropython/micropython/issues/13006#issuecomment-1820309455</a></li>\n<li><a href=\"https://github.com/micropython/micropython/commit/29943546343c92334e8518695a11fc0e2ceea68b\">https://github.com/micropython/micropython/commit/29943546343c92334e8518695a11fc0e2ceea68b</a></li>\n<li><a href=\"https://vuldb.com/?ctiid.277764\">https://vuldb.com/?ctiid.277764</a></li>\n<li><a href=\"https://vuldb.com/?id.277764\">https://vuldb.com/?id.277764</a></li>\n<li><a href=\"https://vuldb.com/?submit.409312\">https://vuldb.com/?submit.409312</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-copy/PYSEC-2024-91.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-copy/PYSEC-2024-91.yaml</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-io/PYSEC-2024-93.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-io/PYSEC-2024-93.yaml</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-os-path/PYSEC-2024-95.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-os-path/PYSEC-2024-95.yaml</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-string/PYSEC-2024-96.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-string/PYSEC-2024-96.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-74qm-4v7r-jw2f\">https://github.com/advisories/GHSA-74qm-4v7r-jw2f</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-74qm-4v7r-jw2f",
            "title": "[micropython-string] Heap-based Buffer Overflow in MicroPython",
            "date_modified": "2024-09-25T18:23:42.000Z",
            "date_published": "2024-09-17T21:30:33.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-74qm-4v7r-jw2f"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A vulnerability was found in MicroPython 1.23.0. It has been classified as critical. Affected is the function mp_vfs_umount of the file extmod/vfs.c of the component VFS Unmount Handler. The manipulation leads to heap-based buffer overflow. It is possible to launch the attack remotely. The exploit has been disclosed to the public and may be used. The name of the patch is 29943546343c92334e8518695a11fc0e2ceea68b. It is recommended to apply a patch to fix this issue. In the VFS unmount process, the comparison between the mounted path string and the unmount requested string is based solely on the length of the unmount string, which can lead to a heap buffer overflow read.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-8946\">https://nvd.nist.gov/vuln/detail/CVE-2024-8946</a></li>\n<li><a href=\"https://github.com/micropython/micropython/issues/13006\">https://github.com/micropython/micropython/issues/13006</a></li>\n<li><a href=\"https://github.com/micropython/micropython/issues/13006#issuecomment-1820309455\">https://github.com/micropython/micropython/issues/13006#issuecomment-1820309455</a></li>\n<li><a href=\"https://github.com/micropython/micropython/commit/29943546343c92334e8518695a11fc0e2ceea68b\">https://github.com/micropython/micropython/commit/29943546343c92334e8518695a11fc0e2ceea68b</a></li>\n<li><a href=\"https://vuldb.com/?ctiid.277764\">https://vuldb.com/?ctiid.277764</a></li>\n<li><a href=\"https://vuldb.com/?id.277764\">https://vuldb.com/?id.277764</a></li>\n<li><a href=\"https://vuldb.com/?submit.409312\">https://vuldb.com/?submit.409312</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-copy/PYSEC-2024-91.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-copy/PYSEC-2024-91.yaml</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-io/PYSEC-2024-93.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-io/PYSEC-2024-93.yaml</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-os-path/PYSEC-2024-95.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-os-path/PYSEC-2024-95.yaml</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-string/PYSEC-2024-96.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-string/PYSEC-2024-96.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-74qm-4v7r-jw2f\">https://github.com/advisories/GHSA-74qm-4v7r-jw2f</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-74qm-4v7r-jw2f",
            "title": "[micropython-os-path] Heap-based Buffer Overflow in MicroPython",
            "date_modified": "2024-09-25T18:23:42.000Z",
            "date_published": "2024-09-17T21:30:33.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-74qm-4v7r-jw2f"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A vulnerability was found in MicroPython 1.23.0. It has been classified as critical. Affected is the function mp_vfs_umount of the file extmod/vfs.c of the component VFS Unmount Handler. The manipulation leads to heap-based buffer overflow. It is possible to launch the attack remotely. The exploit has been disclosed to the public and may be used. The name of the patch is 29943546343c92334e8518695a11fc0e2ceea68b. It is recommended to apply a patch to fix this issue. In the VFS unmount process, the comparison between the mounted path string and the unmount requested string is based solely on the length of the unmount string, which can lead to a heap buffer overflow read.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-8946\">https://nvd.nist.gov/vuln/detail/CVE-2024-8946</a></li>\n<li><a href=\"https://github.com/micropython/micropython/issues/13006\">https://github.com/micropython/micropython/issues/13006</a></li>\n<li><a href=\"https://github.com/micropython/micropython/issues/13006#issuecomment-1820309455\">https://github.com/micropython/micropython/issues/13006#issuecomment-1820309455</a></li>\n<li><a href=\"https://github.com/micropython/micropython/commit/29943546343c92334e8518695a11fc0e2ceea68b\">https://github.com/micropython/micropython/commit/29943546343c92334e8518695a11fc0e2ceea68b</a></li>\n<li><a href=\"https://vuldb.com/?ctiid.277764\">https://vuldb.com/?ctiid.277764</a></li>\n<li><a href=\"https://vuldb.com/?id.277764\">https://vuldb.com/?id.277764</a></li>\n<li><a href=\"https://vuldb.com/?submit.409312\">https://vuldb.com/?submit.409312</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-copy/PYSEC-2024-91.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-copy/PYSEC-2024-91.yaml</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-io/PYSEC-2024-93.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-io/PYSEC-2024-93.yaml</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-os-path/PYSEC-2024-95.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-os-path/PYSEC-2024-95.yaml</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-string/PYSEC-2024-96.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-string/PYSEC-2024-96.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-74qm-4v7r-jw2f\">https://github.com/advisories/GHSA-74qm-4v7r-jw2f</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-74qm-4v7r-jw2f",
            "title": "[micropython-io] Heap-based Buffer Overflow in MicroPython",
            "date_modified": "2024-09-25T18:23:42.000Z",
            "date_published": "2024-09-17T21:30:33.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-74qm-4v7r-jw2f"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A vulnerability was found in MicroPython 1.23.0. It has been classified as critical. Affected is the function mp_vfs_umount of the file extmod/vfs.c of the component VFS Unmount Handler. The manipulation leads to heap-based buffer overflow. It is possible to launch the attack remotely. The exploit has been disclosed to the public and may be used. The name of the patch is 29943546343c92334e8518695a11fc0e2ceea68b. It is recommended to apply a patch to fix this issue. In the VFS unmount process, the comparison between the mounted path string and the unmount requested string is based solely on the length of the unmount string, which can lead to a heap buffer overflow read.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-8946\">https://nvd.nist.gov/vuln/detail/CVE-2024-8946</a></li>\n<li><a href=\"https://github.com/micropython/micropython/issues/13006\">https://github.com/micropython/micropython/issues/13006</a></li>\n<li><a href=\"https://github.com/micropython/micropython/issues/13006#issuecomment-1820309455\">https://github.com/micropython/micropython/issues/13006#issuecomment-1820309455</a></li>\n<li><a href=\"https://github.com/micropython/micropython/commit/29943546343c92334e8518695a11fc0e2ceea68b\">https://github.com/micropython/micropython/commit/29943546343c92334e8518695a11fc0e2ceea68b</a></li>\n<li><a href=\"https://vuldb.com/?ctiid.277764\">https://vuldb.com/?ctiid.277764</a></li>\n<li><a href=\"https://vuldb.com/?id.277764\">https://vuldb.com/?id.277764</a></li>\n<li><a href=\"https://vuldb.com/?submit.409312\">https://vuldb.com/?submit.409312</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-copy/PYSEC-2024-91.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-copy/PYSEC-2024-91.yaml</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-io/PYSEC-2024-93.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-io/PYSEC-2024-93.yaml</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-os-path/PYSEC-2024-95.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-os-path/PYSEC-2024-95.yaml</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-string/PYSEC-2024-96.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-string/PYSEC-2024-96.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-74qm-4v7r-jw2f\">https://github.com/advisories/GHSA-74qm-4v7r-jw2f</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-74qm-4v7r-jw2f",
            "title": "[micropython-copy] Heap-based Buffer Overflow in MicroPython",
            "date_modified": "2024-09-25T18:23:42.000Z",
            "date_published": "2024-09-17T21:30:33.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-74qm-4v7r-jw2f"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>Multipart file upload support as defined in the <a href=\"https://github.com/jaydenseric/graphql-multipart-request-spec\">GraphQL multipart request specification</a> was enabled by default in all Strawberry HTTP view integrations. This made all Strawberry HTTP view integrations vulnerable to CSRF attacks if users did not explicitly enable CSRF preventing security mechanism for their servers.\nAdditionally, the Django HTTP view integration, in particular, had an exemption for Django's built-in CSRF protection (i.e., the <code>CsrfViewMiddleware</code> middleware) by default.</p>\n<p>In affect, all Strawberry integrations were vulnerable to CSRF attacks by default.</p>\n<h3 id=\"patches\">Patches</h3>\n<p>Version <code>v0.243.0</code> is the first <code>strawberry-graphql</code> including a patch. Check out our <a href=\"https://strawberry.rocks/docs/breaking-changes/0.243.0\">documentation</a> for additional details and upgrade instructions.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://strawberry.rocks/docs/breaking-changes/0.243.0\">Strawberry upgrade guide</a></li>\n<li><a href=\"https://github.com/jaydenseric/graphql-multipart-request-spec/blob/master/readme.md#security\">Multipart Upload Security Implications</a></li>\n</ul>\n<h3 id=\"credits\">Credits</h3>\n<ul>\n<li><a href=\"https://github.com/graingert\">Thomas Grainger</a></li>\n<li><a href=\"https://github.com/speedy1991\">Arthur Bayr</a></li>\n<li><a href=\"https://github.com/DoctorJohn\">Jonathan Ehwald</a></li>\n</ul>\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/strawberry-graphql/strawberry/security/advisories/GHSA-79gp-q4wv-33fr\">https://github.com/strawberry-graphql/strawberry/security/advisories/GHSA-79gp-q4wv-33fr</a></li>\n<li><a href=\"https://github.com/strawberry-graphql/strawberry/commit/37265b230e511480a9ceace492f9f6a484be1387\">https://github.com/strawberry-graphql/strawberry/commit/37265b230e511480a9ceace492f9f6a484be1387</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-47082\">https://nvd.nist.gov/vuln/detail/CVE-2024-47082</a></li>\n<li><a href=\"https://strawberry.rocks/docs/breaking-changes/0.243.0\">https://strawberry.rocks/docs/breaking-changes/0.243.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-79gp-q4wv-33fr\">https://github.com/advisories/GHSA-79gp-q4wv-33fr</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-79gp-q4wv-33fr",
            "title": "[strawberry-graphql] Cross-Site Request Forgery (CSRF) in strawberry-graphql",
            "date_modified": "2024-09-25T19:30:16.000Z",
            "date_published": "2024-09-25T18:21:19.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-79gp-q4wv-33fr"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A vulnerability was found in MicroPython 1.22.2. It has been declared as critical. Affected by this vulnerability is an unknown functionality of the file py/objarray.c. The manipulation leads to use after free. The attack can be launched remotely. The complexity of an attack is rather high. The exploitation appears to be difficult. Upgrading to version 1.23.0 is able to address this issue. The identifier of the patch is 4bed614e707c0644c06e117f848fa12605c711cd. It is recommended to upgrade the affected component. In micropython objarray component, when a bytes object is resized and copied into itself, it may reference memory that has already been freed.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-8947\">https://nvd.nist.gov/vuln/detail/CVE-2024-8947</a></li>\n<li><a href=\"https://github.com/micropython/micropython/issues/13283\">https://github.com/micropython/micropython/issues/13283</a></li>\n<li><a href=\"https://github.com/micropython/micropython/issues/13283#issuecomment-1918479709\">https://github.com/micropython/micropython/issues/13283#issuecomment-1918479709</a></li>\n<li><a href=\"https://github.com/micropython/micropython/commit/4bed614e707c0644c06e117f848fa12605c711cd\">https://github.com/micropython/micropython/commit/4bed614e707c0644c06e117f848fa12605c711cd</a></li>\n<li><a href=\"https://github.com/micropython/micropython/releases/tag/v1.23.0\">https://github.com/micropython/micropython/releases/tag/v1.23.0</a></li>\n<li><a href=\"https://vuldb.com/?ctiid.277765\">https://vuldb.com/?ctiid.277765</a></li>\n<li><a href=\"https://vuldb.com/?id.277765\">https://vuldb.com/?id.277765</a></li>\n<li><a href=\"https://vuldb.com/?submit.409316\">https://vuldb.com/?submit.409316</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-copy/PYSEC-2024-92.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-copy/PYSEC-2024-92.yaml</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-io/PYSEC-2024-94.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-io/PYSEC-2024-94.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-pwwp-3q7j-9mx8\">https://github.com/advisories/GHSA-pwwp-3q7j-9mx8</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-pwwp-3q7j-9mx8",
            "title": "[micropython-io] Use After Free in MicroPython",
            "date_modified": "2024-09-25T18:18:17.000Z",
            "date_published": "2024-09-17T21:30:32.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-pwwp-3q7j-9mx8"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A vulnerability was found in MicroPython 1.22.2. It has been declared as critical. Affected by this vulnerability is an unknown functionality of the file py/objarray.c. The manipulation leads to use after free. The attack can be launched remotely. The complexity of an attack is rather high. The exploitation appears to be difficult. Upgrading to version 1.23.0 is able to address this issue. The identifier of the patch is 4bed614e707c0644c06e117f848fa12605c711cd. It is recommended to upgrade the affected component. In micropython objarray component, when a bytes object is resized and copied into itself, it may reference memory that has already been freed.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-8947\">https://nvd.nist.gov/vuln/detail/CVE-2024-8947</a></li>\n<li><a href=\"https://github.com/micropython/micropython/issues/13283\">https://github.com/micropython/micropython/issues/13283</a></li>\n<li><a href=\"https://github.com/micropython/micropython/issues/13283#issuecomment-1918479709\">https://github.com/micropython/micropython/issues/13283#issuecomment-1918479709</a></li>\n<li><a href=\"https://github.com/micropython/micropython/commit/4bed614e707c0644c06e117f848fa12605c711cd\">https://github.com/micropython/micropython/commit/4bed614e707c0644c06e117f848fa12605c711cd</a></li>\n<li><a href=\"https://github.com/micropython/micropython/releases/tag/v1.23.0\">https://github.com/micropython/micropython/releases/tag/v1.23.0</a></li>\n<li><a href=\"https://vuldb.com/?ctiid.277765\">https://vuldb.com/?ctiid.277765</a></li>\n<li><a href=\"https://vuldb.com/?id.277765\">https://vuldb.com/?id.277765</a></li>\n<li><a href=\"https://vuldb.com/?submit.409316\">https://vuldb.com/?submit.409316</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-copy/PYSEC-2024-92.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-copy/PYSEC-2024-92.yaml</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/micropython-io/PYSEC-2024-94.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/micropython-io/PYSEC-2024-94.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-pwwp-3q7j-9mx8\">https://github.com/advisories/GHSA-pwwp-3q7j-9mx8</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-pwwp-3q7j-9mx8",
            "title": "[micropython-copy] Use After Free in MicroPython",
            "date_modified": "2024-09-25T18:18:17.000Z",
            "date_published": "2024-09-17T21:30:32.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-pwwp-3q7j-9mx8"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p><em>What kind of vulnerability is it? Who is impacted?</em></p>\n<p>Open redirect vulnerability - a maliciously crafted link to a notebook server could redirect the browser to a different website.</p>\n<p>All notebook servers are technically affected, however, these maliciously crafted links can only be reasonably made for known notebook server hosts. A link to your notebook server may <em>appear</em> safe, but ultimately redirect to a spoofed server on the public internet.</p>\n<h3 id=\"patches\">Patches</h3>\n<p><em>Has the problem been patched? What versions should users upgrade to?</em></p>\n<p>Patched in notebook 6.1.5</p>\n<h3 id=\"references\">References</h3>\n<p><a href=\"https://cheatsheetseries.owasp.org/cheatsheets/Unvalidated_Redirects_and_Forwards_Cheat_Sheet.html\">OWASP page on open redirects</a></p>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>If you have any questions or comments about this advisory, or vulnerabilities to report, please email our security list <a href=\"mailto:security@ipython.org\">security@ipython.org</a>.</p>\n<p>Credit: zhuonan li of Alibaba Application Security Team</p>\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/jupyter/notebook/security/advisories/GHSA-c7vm-f5p4-8fqh\">https://github.com/jupyter/notebook/security/advisories/GHSA-c7vm-f5p4-8fqh</a></li>\n<li><a href=\"https://github.com/jupyter/notebook/commit/3cec4bbe21756de9f0c4bccf18cf61d840314d74\">https://github.com/jupyter/notebook/commit/3cec4bbe21756de9f0c4bccf18cf61d840314d74</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2020-26215\">https://nvd.nist.gov/vuln/detail/CVE-2020-26215</a></li>\n<li><a href=\"https://lists.debian.org/debian-lts-announce/2020/12/msg00004.html\">https://lists.debian.org/debian-lts-announce/2020/12/msg00004.html</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/notebook/PYSEC-2020-215.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/notebook/PYSEC-2020-215.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-c7vm-f5p4-8fqh\">https://github.com/advisories/GHSA-c7vm-f5p4-8fqh</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-c7vm-f5p4-8fqh",
            "title": "[notebook] Open redirect in Jupyter Notebook",
            "date_modified": "2024-09-25T17:58:44.000Z",
            "date_published": "2020-11-18T21:06:36.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-c7vm-f5p4-8fqh"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>Users of the MLflow Open Source Project who are hosting the MLflow Model Registry using the <code>mlflow server</code> or <code>mlflow ui</code> commands using an MLflow version older than MLflow 2.2.1 may be vulnerable to a remote file existence check exploit if they are not limiting who can query their server (for example, by using a cloud VPC, an IP allowlist for inbound requests, or authentication / authorization middleware).</p>\n<p>This issue only affects users and integrations that run the <code>mlflow server</code> and <code>mlflow ui</code> commands. Integrations that do not make use of <code>mlflow server</code> or <code>mlflow ui</code> are unaffected; for example, the Databricks Managed MLflow product and MLflow on Azure Machine Learning do not make use of these commands and are not impacted by these vulnerabilities in any way.</p>\n<p>The vulnerability detailed in <a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-1176\">https://nvd.nist.gov/vuln/detail/CVE-2023-1176</a> enables an actor to check the existence of arbitrary files unrelated to MLflow from the host server, including any files stored in remote locations to which the host server has access.</p>\n<h3 id=\"patches\">Patches</h3>\n<p>This vulnerability has been patched in MLflow 2.2.1, which was released to PyPI on March 2nd, 2023. If you are using <code>mlflow server</code> or <code>mlflow ui</code> with the MLflow Model Registry, we recommend upgrading to MLflow 2.2.1 as soon as possible.</p>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p>If you are using the MLflow open source <code>mlflow server</code> or <code>mlflow ui</code> commands, we strongly recommend limiting who can access your MLflow Model Registry and MLflow Tracking servers using a cloud VPC, an IP allowlist for inbound requests, authentication / authorization middleware, or another access restriction mechanism of your choosing.</p>\n<p>If you are using the MLflow open source <code>mlflow server</code> or <code>mlflow ui</code> commands, we also strongly recommend limiting the remote files to which your MLflow Model Registry and MLflow Tracking servers have access. For example, if your MLflow Model Registry or MLflow Tracking server uses cloud-hosted blob storage for MLflow artifacts, make sure to restrict the scope of your server's cloud credentials such that it can only access files and directories related to MLflow.</p>\n<h3 id=\"references\">References</h3>\n<p>More information about the vulnerability is available at <a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-1176\">https://nvd.nist.gov/vuln/detail/CVE-2023-1176</a>.</p>\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/mlflow/mlflow/security/advisories/GHSA-wp72-7hj9-5265\">https://github.com/mlflow/mlflow/security/advisories/GHSA-wp72-7hj9-5265</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-1176\">https://nvd.nist.gov/vuln/detail/CVE-2023-1176</a></li>\n<li><a href=\"https://github.com/mlflow/mlflow/commit/63ef72aa4334a6473ce7f889573c92fcae0b3c0d\">https://github.com/mlflow/mlflow/commit/63ef72aa4334a6473ce7f889573c92fcae0b3c0d</a></li>\n<li><a href=\"https://huntr.dev/bounties/ae92f814-6a08-435c-8445-eec0ef4f1085\">https://huntr.dev/bounties/ae92f814-6a08-435c-8445-eec0ef4f1085</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/mlflow/PYSEC-2023-28.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/mlflow/PYSEC-2023-28.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-wp72-7hj9-5265\">https://github.com/advisories/GHSA-wp72-7hj9-5265</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-wp72-7hj9-5265",
            "title": "[mlflow] Remote file existence check vulnerability in `mlflow server` and `mlflow ui` CLIs",
            "date_modified": "2024-09-25T17:51:10.000Z",
            "date_published": "2023-03-24T22:01:15.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-wp72-7hj9-5265"
            },
            "tags": [
                "severity"
            ]
        }
    ]
}